### PR TITLE
Input deeper dive: Input events

### PR DESCRIPTION
**Pull request by greym0uth**
## Description

There wasn't really much reference to using event channels when you need greater control over input events that the `InputHandler` doesn't cover, so I added a section with a brief example on listening to the `InputEvent` event channel.

## Additions

- Added input event channel section (under input section) to book.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Added unit tests for new code added in this PR.
- [ ] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Updated the content of the book if this PR would make the book outdated.

If this modified or created any rs files:

- [ ] Ran `cargo +stable fmt --all`
- [ ] Ran `cargo clippy --all --features "empty"` (may require `cargo clean` before)
- [ ] Ran `cargo build --features "empty"`
- [ ] Ran `cargo test --all --features "empty"`

### CODE REVIEW

File: book/src/input/watching_input_events.md
Comment (CleanCut) [inline book/src/input/watching_input_events.md:40]:
```suggestion
Setting up the system has extra requirements. You will need to store a `ReaderId` in your system struct used to read the type of input events you care about. In this example, our struct field is called `input_event_rid` and the type of event we are listening to is `InputEvent<StringBindings>`, but you would use whatever event type you would want to use in your own game.  You need to initialize your `ReaderID` in the system's `new` method (see the [event channel](../concepts/event-channel.md) guide for more details and another example of doing this).
```

File: book/src/input/watching_input_events.md
Comment (CleanCut) [inline book/src/input/watching_input_events.md:29]: I suggest adding a newline between `type SystemData...` and `fn run(...` for readability.
```suggestion
    type SystemData = Read<'a, EventChannel<InputEvent<StringBindings>>>;
    
```

File: book/src/input/watching_input_events.md
Comment (CleanCut) [inline book/src/input/watching_input_events.md:42]:
```suggestion
Next let's take a look at our implementation of the `System` trait. We set the `SystemData` associated type to a `Read` storage for the type of event channel we want.  In the `run` method we iterate over all the input events the occured. Take a look at the [`InputEvent`][doc_input_event] enum for all the available input events you can handle.
```

File: book/src/input/watching_input_events.md
Comment (CleanCut) [inline book/src/input/watching_input_events.md:44]:
```suggestion
There you have it! Now you can read input events as they occur. Be sure to look at the [`event channel`](../concepts/event-channel.md) section for a better guide on using event channels and enjoy listening to input events!
```

**Review by CleanCut — CHANGES_REQUESTED** [review]
I really love this example.  Thank you!

To Do:
- A new page in the book is worthy of a [change log entry](https://github.com/amethyst/amethyst/blob/master/docs/CHANGELOG.md), please go ahead and add one!
- I have several suggestions for making the wording more clear.  You don't have to accept these all as-is (though you certainly can) if you have other ideas for how to better clarify, but it took me making these changes to be able to understand what was being said.

**Comment by CleanCut** [comment]
This looks good to me, but I'm going to hold off approving/merging until the legion work has been merged in so as to not add more burden to that effort.

**Comment by Jaden-Giordano** [comment]
Does this overlap with that effort? If so, is this something that would be finished before the end of this month. I know some people may come to this repo for hacktoberfest but might be put if any PR they submit for that doesn't get included in their contributions.

**Comment by jojolepro** [comment]
@Jaden-Giordano We are accepting pull requests again now that legion got merged. Feel free to update this pull request to the latest version of the repository and we'll merge it. Thanks!

**Comment by jojolepro** [comment]
Since we moved to legion and  this example still uses specs, we'll need to update that. We will keep this PR opened to make sure we don't forget about it while updating the book. If you have time, you can update the code to use legion, which will save us some time. Thanks! :hearts:

**Pull request closed without merging**
